packages/amplify_api/example/integration_test/graphql_tests.dart

```diff
@@ -313,6 +313,29 @@ void main() {
             ModelMutations.deleteById(Comment.classType, createdComment.id);
         await Amplify.API.mutate(request: deleteCommentReq).response;
       });
+
+      testWidgets('should decode a custom list request',
+          (WidgetTester tester) async {
+        final name = 'Lorem Ipsum Test Blog: ${UUID.getUUID()}';
+        await addBlog(name);
+
+        const listBlogs = 'listBlogs';
+        String graphQLDocument = '''query GetBlogsCustomDecoder {
+          $listBlogs {
+            items {
+              id
+              name
+              createdAt
+            }
+          }
+        }''';
+        final request = GraphQLRequest<PaginatedResult<Blog>>(
+            document: graphQLDocument,
+            modelType: const PaginatedModelTypeImpl(Blog.classType),
```

Comment:
Callout: when we named this class we did not intend for it to be developer-facing as we were not considering the customer request use case very much. Now that we are exposing it like this, don't know if we like a different name?

Reply:
I don't have enough context to suggest a name, but I would vote for renaming it.

Are we expecting there to be other classes that extend `PaginatedModelType`? Do we need `PaginatedModelType` and `PaginatedModelTypeImpl`?

Reply:
What about:

```dart
extension PaginatedModelTypeExt<T extends Model> on ModelType<T> {
  PaginatedModelType<T> get list => PaginatedModelTypeImpl(this);
}

// so it becomes

GraphQLRequest<PaginatedResult<Blog>>(
  document: graphQLDocument,
  modelType: Blog.classType.list,
  // ...
```

This way the class can stay private and it's more in line with the current approach.

Reply:
In latest revision refactored to follow @Jordan-Nelson 's suggestion here.